core/string/ustring.cpp

```diff
@@ -3582,6 +3582,51 @@ String String::dedent() const {
 	return new_string;
 }
 
+String String::strip_bbcode() const {
+	String new_string;
+	for (int i = 0; i < length(); i++) {
+		bool found_close = false;
+
+		// Checks for "["
+		if (operator[](i) == '[') {
+			int skip_char = 0;
+
+			//Check for [rb] and [lb].
```

Comment:
I guess `strip_bbcode` should just remove all BBCodes without parsing them (even `[code]`). That is:

```gdscript
"[lb]text[rb]".strip_bbcode() # "text", not "[text]"
```

Otherwise, after `strip_bbcode`, we can get a string containing the BBCode, which even sounds wrong:

```gdscript
"[lb]b[rb]text[lb]/b[rb]".strip_bbcode() # "[b]text[/b]"
```

As I wrote above, if the functionality is needed, then it should not be the `String.strip_bbcode` method, but `RichTextLabel.get_parsed_text_from_string` or something similar.

Reply:
But what if they want brackets in the text they're typing for some reason? I don't see why example 2 would ever come up outside of somebody trying to intentionally "break" it.